pysat/tests/test_meta.py

```diff
@@ -15,93 +15,113 @@
+
+        assert 'children' not in self.meta.data.columns
+        assert self.dval not in self.meta.keys_nD()
+
+    def test_meta_repr(self):
+        """ Test the Meta repr function
+        """
+        self.out = self.meta.__repr__()
+        assert isinstance(self.out, str)
+        assert self.out.find('Meta(') >= 0
 
     def test_setting_nonpandas_metadata(self):
+        """ Test meta initialization with bad metadata
+        """
         with pytest.raises(ValueError):
             self.meta = pysat.Meta(metadata='Not a Panda')
 
-    def test_inst_data_assign_meta_default(self):
-        self.testInst.load(2009, 1)
-        self.testInst['help'] = self.testInst['mlt']
-
-        assert self.testInst.meta['help', 'long_name'] == 'help'
-        assert self.testInst.meta['help', 'axis'] == 'help'
-        assert self.testInst.meta['help', 'label'] == 'help'
-        assert self.testInst.meta['help', 'notes'] == ''
-        assert np.isnan(self.testInst.meta['help', 'fill'])
-        assert np.isnan(self.testInst.meta['help', 'value_min'])
-        assert np.isnan(self.testInst.meta['help', 'value_max'])
-        assert self.testInst.meta['help', 'units'] == ''
-        assert self.testInst.meta['help', 'desc'] == ''
-        assert self.testInst.meta['help', 'scale'] == 'linear'
-
-    def test_inst_data_assign_meta(self):
-        self.testInst.load(2009, 1)
-        self.testInst['help'] = {'data': self.testInst['mlt'],
-                                 'units': 'V',
-                                 'long_name': 'The Doors'}
-        assert self.testInst.meta['help', 'long_name'] == 'The Doors'
-        assert self.testInst.meta['help', 'axis'] == 'help'
-        assert self.testInst.meta['help', 'label'] == 'help'
-        assert self.testInst.meta['help', 'notes'] == ''
-        assert np.isnan(self.testInst.meta['help', 'fill'])
-        assert np.isnan(self.testInst.meta['help', 'value_min'])
-        assert np.isnan(self.testInst.meta['help', 'value_max'])
-        assert self.testInst.meta['help', 'units'] == 'V'
-        assert self.testInst.meta['help', 'desc'] == ''
-        assert self.testInst.meta['help', 'scale'] == 'linear'
-
-    def test_inst_data_assign_meta_empty_list(self):
-        self.testInst.load(2009, 1)
-        self.testInst['help'] = {'data': self.testInst['mlt'],
-                                 'units': [],
-                                 'long_name': 'The Doors'}
-        assert self.testInst.meta['help', 'units'] == ''
-
-    def test_inst_data_assign_meta_string_list(self):
-        self.testInst.load(2009, 1)
-        self.testInst['help'] = {'data': self.testInst['mlt'],
-                                 'units': ['A', 'B'],
-                                 'long_name': 'The Doors'}
-        assert self.testInst.meta['help', 'units'] == 'A\n\nB'
+    @pytest.mark.parametrize("set_dict",
+                             [({}), ({'units': 'V', 'long_name': 'Longgggg'})])
```

Comment:
Looks like the cause of the bug is that meta expects units to be a list of strings not a string.  This doesn't match with what the tests were doing before, but the problem could be in meta itself.
```
AssertionError: assert 'V' == ['V']
```

Reply:
I've been working on this, and it turns out that parametrize doesn't like dict inputs.  But only fails some of the time.